RecoTracker/TkSeedGenerator/plugins/SeedFitter.cc

```diff
@@ -0,0 +1,210 @@
+#include "SeedFitter.h"
+
+#include "TrackingTools/KalmanUpdators/interface/KFUpdator.h"
+#include "RecoTracker/TkSeedGenerator/interface/FastHelix.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Utilities/interface/ESInputTag.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateTransform.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateOnSurface.h"
+#include "FWCore/Utilities/interface/Likely.h"
+
+namespace {
+
+  template <class T>
+  inline T sqr(T t) {
+    return t * t;
+  }
+
+}  // namespace
+
+SeedFitter::SeedFitter(const edm::ParameterSet& cfg) : SeedFitter::SeedFitter(cfg, consumesCollector()) {}
+
+SeedFitter::SeedFitter(const edm::ParameterSet& cfg, edm::ConsumesCollector&& iC)
+    : eleSeedCollectionToken_(
+          consumes<reco::ElectronSeedCollection>(cfg.getParameter<edm::InputTag>("eleSeedCollection"))),
+      thePropagatorLabel_(cfg.getParameter<std::string>("propagator")),
+      theBOFFMomentum_(cfg.getParameter<double>("SeedMomentumForBOFF")),
+      theOriginTransverseErrorMultiplier_(cfg.getParameter<double>("OriginTransverseErrorMultiplier")),
+      theMinOneOverPtError_(cfg.getParameter<double>("MinOneOverPtError")),
+      TTRHBuilder_(cfg.getParameter<std::string>("TTRHBuilder")),
+      mfName_(cfg.getParameter<std::string>("magneticField")),
+      trackerGeometryESToken_(iC.esConsumes()),
+      propagatorESToken_(iC.esConsumes(edm::ESInputTag("", thePropagatorLabel_))),
+      magneticFieldESToken_(iC.esConsumes(edm::ESInputTag("", mfName_))),
+      transientTrackingRecHitBuilderESToken_(iC.esConsumes(edm::ESInputTag("", TTRHBuilder_))) {
+  produces<reco::ElectronSeedCollection>();
+}
+
+SeedFitter::~SeedFitter() {}
+
+void SeedFitter::fillDescription(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+
+  desc.add<edm::InputTag>("eleSeedCollection", edm::InputTag("hltEgammaFittedElectronPixelSeeds"));
+  desc.add<std::string>("propagator", "PropagatorWithMaterialParabolicMf");
+  desc.add<double>("SeedMomentumForBOFF", 5.0);
+  desc.add<double>("OriginTransverseErrorMultiplier", 1.0);
+  desc.add<double>("MinOneOverPtError", 1.0);
+  desc.add<std::string>("TTRHBuilder", "WithTrackAngle");
+  desc.add<std::string>("magneticField", "ParabolicMf");
+
+  descriptions.add("seedFitter", desc);
```

Comment:
```suggestion
  descriptions.addWithDefaultLabel(desc); 
```

slightly more idiomatic (with the same outcome).